src/dsm/headers/Node.hpp

```diff
@@ -313,13 +318,14 @@ namespace dsm {
   template <typename Id, typename Size, typename Delay>
     requires(std::unsigned_integral<Id> && std::unsigned_integral<Size> &&
              std::unsigned_integral<Delay>)
-  TrafficLight<Id, Size, Delay>::TrafficLight(Id id) : Node<Id, Size>{id}, m_counter{0} {}
+  TrafficLight<Id, Size, Delay>::TrafficLight(Id id)
+      : Node<Id, Size>{id}, m_counter{0}, m_phase{0} {}
 
   template <typename Id, typename Size, typename Delay>
     requires(std::unsigned_integral<Id> && std::unsigned_integral<Size> &&
              std::unsigned_integral<Delay>)
   TrafficLight<Id, Size, Delay>::TrafficLight(Node<Id, Size> node)
-      : Node<Id, Size>{node.id()}, m_counter{0} {
+      : Node<Id, Size>{node.id()}, m_counter{0}, m_phase{0} {
     this->setCoords(node.coords());
     this->setCapacity(node.capacity());
```

Comment:
Maybe I'm forgetting something, but this node is passed neither by move nor by const ref.

Reply:
It's the current behaviour... but you got the point, I'm going to change it

Reply:
Ok that's a big NOPE beacuse the incoming branch is not mine. @LudovicoLudovico can you change the TL constructor in order to take a const ref of a node instead of a copy?